### PR TITLE
Update process.md about reviewer assignment #4826

### LINKED ISSUE #4826: Update process.md about reviewer assignment
i.e. how to choose reviewer, assign milestone

### PR DESCRIPTION
**Pull request by damithc**
Fixes #4826

### CODE REVIEW

**Comment by damithc** [comment]
I feel this bit of workflow is a bit complicated and should be handled by the release lead instead.

File: devdocs/process.md [hunk 1]
```diff
@@ -107,9 +107,25 @@ This workflow is an adaptation of the [GitHub flow](https://guides.github.com/in
...
+        * Assign a reviewer for the PR. The selection of reviewer
+          should be based on the *feature* (`f-`), *aspect* (`a-`), or the *tech* (`t-`) labels 
+          of the issue. If there are more than one of these labels, the main category of
+          the issue as specified by the *main classifier* (`m.`) label. <br>
+          e.g. If the issue has labels `m-Tech` `a-Testing` `t-JS`, the main classifier is 
+          `Tech` and therefore the reviewer should be based on the `t-JS` label.<br>
+          Once you have figured out which label determines the reviewer, you can find the 
+          *Lead* of the label in the [Issue Labels section](#issue-labels).
+        * Assign a milestone to the PR. Usually, you can choose between 
+          of the upcoming milestone or the one after. This is to encourage you to finish
```
Comment (thyageshm) [inline devdocs/process.md]: I think we can remove "of"

**Comment by thyageshm** [comment]
@damithc sir, which part of the workflow should be handled by the release lead?

**Comment by damithc** [comment]
This new part. i.e. assigning reviewers and milestones for incoming PRs. Can be done once per day. We can do some load balancing if we do it centrally e.g. if someone is overloaded or busy, we can assign someone else.

**Comment by damithc** [comment]
It doesn't have to be release lead, but it should be a specific person (rather than 'everybody').

**Comment by junhaoyap** [comment]
This is indeed quite complicated and it seems like it could result in overloading of some people during certain periods of time - I reckon some load balancing is needed as well

**Comment by thyageshm** [comment]
could this be the area leads? if an issue is under their area, they have to assign someone. If they dont assign, then the release lead can assign them to the task

**Comment by damithc** [comment]
The more we spread the responsibility, the more people I have to chase. Area leads are unproven, so I don't want to give them too much responsibility until they are proven. Their responsibility is to review issues and fix urgent issues in their area on a timely manner. If they do that well, they can become project leads and take more responsibility. 

We have three project leads and a PM, so the command and control work need not be pushed to junior members. We can divide the work among the four of us in a clear way. 

This is what I suggest:

For incoming issues:
* PM will assign a priority
* If important labels are missing, PM will add those too.

For incoming PRs:
* PL will assign a reviewer based on issue labels and current workload of reviewers
* PL will assign the current+1 milestone to the issue (this ensures that all new PRs are given at least one week to complete and all ongoing issues have a deadline).

Friday: PL reminds owners of pending issues in the current milestone. They have three more days to finish the issue.

After release: 
* PL do the release
* PL extends the deadline of unfinished issues for one more cycle, or close PR and un-assign issue. 
* Remind those who don't have an issue for the next milestone to take an issue.
* PM deployes

**Comment by damithc** [comment]
Adding current project leads and dev ops area lead to the thread: @LowWeiLin @kanghj @wkurniawan07

**Comment by thyageshm** [comment]
Great! I am in with the process

**Comment by wkurniawan07** [comment]
Sounds good, although the points being discussed go beyond assignment of reviewers (i.e what the issue/PR says).

**Comment by kanghj** [comment]
The process looks fine to me

**Comment by thyageshm** [comment]
@damithc I have a question, if a pr was given V5.71 but is fixes before V5.70 is released, should we keep it as V5.71 for consistency sake?

**Comment by damithc** [comment]
No, we move it to V5.70 in that case.
BTW, I'll revise the PR and the issue title/content soon.

**Comment by thyageshm** [comment]
great

**Comment by damithc** [comment]
Refined further as discussed. I think the PR/issue title is a 'good enough' match although the PR is slightly wider.

File: devdocs/process.md [hunk 2]
```diff
@@ -179,20 +185,57 @@ Role: committer
...
+    * Optionally, apply an `e.` label to the issue (not the PR) to indicate 
+      the estimated effort required to fix the issue.
   * If not green,
     * Delete the merge commit, if any.
     * Change the pull request status to `s.Ongoing`
     * Add a comment to mention the test failure.
-  * Optionally, apply an `e.` label to the issue (not the PR) to indicate 
-    the estimated effort required to fix the issue.
+
+###Assigning lablels, reviewers, and milestones
```
Comment (thyageshm) [inline devdocs/process.md]: labels

File: devdocs/process.md [hunk 3]
```diff
@@ -42,8 +43,7 @@ This workflow is an adaptation of the [GitHub flow](https://guides.github.com/in
    with the latest version of the code from the committer repo.
    ```
    git pull upstream master
-   ```
-6. If you have permissions to change labels, change the issue status to `s.Ongoing`. 
+   ``` 
 
 7. Start a new branch named `{IssueNumber}-{some-keywords}`.
```
Comment (thyageshm) [inline devdocs/process.md]: should we update the numbers? I don't think we HAVE to...

**Comment by thyageshm** [comment]
minor cosmetic changes. The process itself is fine

File: devdocs/process.md (hunk 3 above)
Comment (damithc) [inline devdocs/process.md]: No need. markdown adjusts numbers automatically.

**Comment by damithc** [comment]
Merged after some minor changed. Check the last commit for the changes.

**Pull request merged**